api/accounts/serializers/login.py

```diff
@@ -0,0 +1,22 @@
+from typing import Any, Dict
+
+from rest_framework_simplejwt.serializers import AuthUser, TokenObtainPairSerializer
+from rest_framework_simplejwt.tokens import Token
+
+
+class LoginSerializer(TokenObtainPairSerializer):
+    def validate(self, attrs: Dict[str, Any]) -> Dict[str, str]:
+        return super().validate(
+            {
+                self.username_field: attrs[self.username_field].lower(),
+                "password": attrs["password"],
+            }
+        )
+
+    @classmethod
+    def get_token(cls, user: AuthUser) -> Token:
+        token = super().get_token(user)
+        token["username"] = user.username
+        token["first_name"] = user.first_name
+        token["last_name"] = user.last_name
+        return token
```

Comment:
This function is used to build the Javascript Web Token (JWT) that is used to authenticate the user in future requests. You can use jwt.io to decode the token it generates to see what it looks like.

This function adds the username, first name and last name to the "claims" of the token, so that the front end can easily access these values without needing to request them from the API